state/volume.go

```diff
@@ -750,28 +755,29 @@ func (sb *storageBackend) volumeFilesystemAttachment(host names.Tag, volume name
 	return sb.FilesystemAttachment(host, filesystem.FilesystemTag())
 }
 
-func detachStorageAttachmentOps(host names.Tag, v names.VolumeTag) []txn.Op {
+func detachStorageAttachmentOps(host names.Tag, v names.VolumeTag, asserts bson.D) []txn.Op {
 	id := volumeAttachmentId(host.Id(), v.Id())
 	return []txn.Op{{
 		C:      volumeAttachmentPlanC,
 		Id:     id,
-		Assert: isAliveDoc,
+		Assert: asserts,
 		Update: bson.D{{"$set", bson.D{{"life", Dying}}}},
 	}}
 }
 
-func detachVolumeOps(host names.Tag, v names.VolumeTag) []txn.Op {
+func detachVolumeOps(host names.Tag, v names.VolumeTag, asserts bson.D) []txn.Op {
 	return []txn.Op{{
-		C:      volumeAttachmentsC,
-		Id:     volumeAttachmentId(host.Id(), v.Id()),
-		Assert: isAliveDoc,
+		C:  volumeAttachmentsC,
+		Id: volumeAttachmentId(host.Id(), v.Id()),
+		// If we are forcing...
```

Comment:
May as well flesh out this comment for clarity.

Reply:
cut and paste fail